src/Builder/Visitor/AddPluginToPluginListVisitor.php

```diff
@@ -471,7 +518,16 @@ protected function isPluginAdded(Array_ $node): bool
 
             if (
                 $this->isKeyEqualsToCurrentOne($item)
-                && ($nodeClassName === $classToAdd || in_array($classToAdd, $usedParentClasses))
+                && ($nodeClassName === $classToAdd
+                || in_array($classToAdd, $usedParentClasses))
+                && (
+                    (!$item->value->args
+                    && !count($this->classMetadataTransfer->getConstructorArguments()))
+                    || (
+                        $item->value->args
+                        && $this->isArgumentEqual($item->value->args)
+                    )
+                )
```

Comment:
A lot of check conditions here. Can you please extract them all or partially to the new method with meaningful name? Example: isItemWithoutConstArgs or something like that